File: leetcode/454/454.java
class Solution {
    public int fourSumCount(int[] A, int[] B, int[] C, int[] D) {
        Map<Integer, Integer> m = new HashMap();
        int res = 0;
        for (int a : A)
            for (int b : B)
                m.put(a + b, m.getOrDefault(a + b, 0) + 1);
        for (int c : C)
            for (int d : D)
                res += m.getOrDefault(-c - d, 0);
        return res;
    }
}

File: leetcode/454/454.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
    unordered_map<int, int> m;

  public:
    int fourSumCount(vector<int> &A, vector<int> &B, vector<int> &C,
                     vector<int> &D) {
        m.clear();
        int res = 0;
        for (auto a : A)
            for (auto b : B)
                ++m[a + b];
        for (auto c : C)
            for (auto d : D)
                res += m[-c - d];
        return res;
    }
};